### PR TITLE
Handle nRF Cloud error codes for location services

### PR DESCRIPTION
**Pull request by jayteemo**
Handle new error messages for location service MQTT responses.
Also, handle error codes for REST responses.

NCSDK-13107

### CODE REVIEW

**Comment by jayteemo** [comment]
Draft, docs still need updated.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c [hunk 1]
```diff
@@ -1448,24 +1478,73 @@ static int nrf_cloud_parse_cell_pos_json(const cJSON *const cell_pos_obj,
...
+		return -ENOMSG;
+	}
+
+	ret = get_error_code_value(root_obj, err);
+	if (ret) {
+		goto clean_up;
+	}
+
+	/* If provided, check for matching app id and msg type */
+	if (msg_type &&
```
Comment (plskeggs) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c:1524]: Would it make more sense to check for any provided msg type and app_id before getting the error code above?

Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c:1524]: its a feature!
```
* @retval -ENOENT Error code found, but did not match specified app_id and msg_type.
```

Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c:1524]: OK, I believe you.

Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c:1524]: I didn't look closely enough to see that `err` is a pointer parameter that `get_error_code_value()` updates, and that `ret` is non-zero only if it has trouble actually getting the error code.

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c [hunk 2]
```diff
@@ -98,7 +98,10 @@ int nrf_cloud_cell_pos_process(const char *buf, struct nrf_cloud_cell_pos_result
 	}
 
 	err = nrf_cloud_parse_cell_pos_response(buf, result);
-	if (err < 0) {
```
Comment (junqingzou) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c:101]: This API still returns 1 for No Location Found, can be updated to NRF_CLOUD_ERROR_DATA_NOT_FOUND instead?

**Review by hiltunent — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c (hunk 2 above)
Comment (jayteemo) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_cell_pos.c:101]: a return value of 1 indicates that this data was not a cellular positioning response.
i have updated the function comment to clarify.

`NRF_CLOUD_ERROR_DATA_NOT_FOUND` is an error value sent from nrf cloud.  if the cloud sends such an error, this function returns `-EFAULT`.

**Review by junqingzou — APPROVED** [review]
SLM OK

File: applications/asset_tracker_v2/src/cloud/cloud_codec/nrf_cloud/nrf_cloud_codec.c [hunk 3]
```diff
@@ -464,6 +464,12 @@ int cloud_codec_decode_config(char *input, size_t input_len,
 		return -ENOENT;
 	}
 
+	/* Check for an nRF Cloud message */
+	if ((json_object_decode(root_obj, DATA_GROUP) != NULL) ||
```
Comment (simensrostad) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/nrf_cloud/nrf_cloud_codec.c:468]: Wouldn't this check prevent decoding of incoming shadow messages (config updates)?

Comment (jayteemo) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/nrf_cloud/nrf_cloud_codec.c:468]: this checks for "messageType" or "appId". 
shadow updates are not formatted that way, unless i'm missing something? 
the remainder of `cloud_codec_decode_config` looks for "config" and "state" to handle config updates.

Comment (simensrostad) [reply applications/asset_tracker_v2/src/cloud/cloud_codec/nrf_cloud/nrf_cloud_codec.c:468]: No you're not, I misunderstood the check. That's what you get for reviewing PRs to early in the morning 😄

**Review by simensrostad — APPROVED** [review]
LGTM

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -299,10 +299,10 @@ Libraries for networking
 
     * Updated:
 
-      * During the connection process, shadow data is sent to the application even if no "config" section is present.
-      * The application can now send shadow updates earlier in the connection process.
       * Centralized error handling.
-        Changes to error return values.
+      * Changes to error return values.
+      * For cellular positioning responses, the type is now set based on the "fulfilledWith" response from the nRF Cloud.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:304]:
```suggestion
      * For cellular positioning responses, the type is now set based on the``fulfilledWith`` response from nRF Cloud.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by tokangas — APPROVED** [review]

**Review by peknis — APPROVED** [review]

**Review by jhirsi — APPROVED** [review]
LGTM. BTW, for CI failures, seems that similar failures than in NCSDK-14548 are seen in mosh ci tests of this PR.

**Pull request merged**